scipy/stats/tests/test_stats.py

```diff
@@ -7917,6 +7917,222 @@ def test_alias(self):
+        LB_val = [ x[i][LB[i]] if (not np.isnan(LB[i])) else np.nan 
+                for i in range(m)]
+        UB_val = [ x[i][UB[i]] if (not np.isnan(UB[i])) else np.nan 
+                for i in range(m)]
+        test_val = [(LB_val[i], UB_val[i]) for i in range(m)]
+        # .. Compare to the public function
+        tmp = [
+            stats.quantile_test(x[i], 
+                                0, 
+                                quantile_def[i], 
+                                alternative='two-sided'
+                                ).confidence_interval(confidence[i])
+            for i in range(m)]
+        public = [(tmp[i].low, tmp[i].high) for i in range(m)]
+        assert_equal(public, test_val)
+
+    def test_R_ci_quantile(self):
+        # Using the same test cases as the ci_quantile R function
+        # https://github.com/mayer79/confintr/blob/main/tests/testthat/test-ci_measures_of_location.R#L95
+
+        x = np.exp(np.arange(0, 1.01, 0.01))
+
+        lb, ub = stats.quantile_test(x, 
+                                     0, 
+                                     0.4, 
+                                     alternative='two-sided'
+                                     ).confidence_interval(0.95)
+        expected = [1.349859, 1.648721]
+        assert_almost_equal(expected, [lb,ub], decimal=5)
+
+        lb, ub = stats.quantile_test(x, 
+                                     0, 
+                                     0.5, 
+                                     alternative='two-sided'
+                                     ).confidence_interval(0.9)
+        expected = [1.506818, 1.803988]
+        assert_almost_equal(expected, [lb,ub], decimal=5)
+
+    def test_pval_ci_match(self):
+        # Verify that the following statement hold:
+        #
+        # The 95% confidence interval corresponding with alternative='less' 
+        # has -inf as its lower bound, and if xu is the upper bound, then xu 
+        # is the smallest element from the sample x such that:
+        # `stats.quantile_test(x, q=xu, p=p, alternative='less').pvalue``
+        # will be less than 5%.
+        # 
+        # And the corresponding statement for the alternative='greater' case.
+
+        # This test may fail if we the CI bound falls on a value present 
+        # multiple times in the sample. We make sure this does not happen by 
+        # generating samples via permutations rather than pure random.
+
+        for xi in x:
+            # alternative='less'
+            res = stats.quantile_test(xi, alternative='less')
+            lb, ub = res.confidence_interval()
+            if not np.isnan(ub):
+                i = np.where(xi == ub)[0][0]
+                assert_equal(
+                    stats.quantile_test(xi, 
+                                        q=xi[i], 
+                                        alternative='less'
+                                        ).pvalue <= 0.05, 
+                    True)
```

Comment:
For readability here and elsewhere:
```suggestion
                res = stats.quantile_test(xi, q=xi[i], alternative='less')
                assert_equal(res.pvalue <= 0.05)
```

Reply:
ack